### PR TITLE
Enable support for handling new command line arguments

### LINKED ISSUE #1314: Update python language worker to support handling new command line arguments with functions- prefix
The command-line arguments that the language worker receives from the Functions Host have generic names (such as 'host' and 'port'). This can cause conflicts with other environment variables in certain environments. To address this issue, the Functions Host will now send a new set of arguments prefixed with `functions-<argumentname>`. This will be in addition to the existing arguments.

With the new planned host change, host code will send command line arguments like below. 

> -host 127.0.0.1 --port 50821 --workerId e9efd817-47a1-45dc-9e20-e6f975d7a025 --requestId cbef5957-cdb3-4462-9ee7-ac9f91be0a51 --grpcMaxMessageLength 2147483647 --functions-uri http://127.0.0.1:50821 --functions-workerid e9efd817-47a1-45dc-9e20-e6f975d7a025 --functions-requestid cbef5957-cdb3-4462-9ee7-ac9f91be0a51 --functions-grpcmaxmessagelength 2147483647

The new arguments sent are: 

1. functions-uri
2. functions-request-id
3. functions-worker-id
4. functions-grpc-max-message-length

Initial sanity testing showed that python worker fails to start when the new arguments are sent. We need to fix the python worker to handle these new arguments.

#### Expected behavior with the fix:

1. Use the new arguments (the one with "functions-" prefix).
2. If the new arguments are not present, use the old ones (PORT, HOST etc..)

See https://github.com/Azure/azure-functions-host/issues/9504 for more details.

### PR DESCRIPTION
**Pull request by pdthummar**
## Description
Updated python language worker to support handling new command line arguments with functions- prefix. Verified changes with Host debugging.

Fixes #1314 
---

### PR information

- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] New Unit tests were added for the changes made and CI is passing.

### Quality of Code and Contribution Guidelines
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-functions-python-worker/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: azure_functions_worker/main.py [hunk 1]
```diff
@@ -26,6 +26,17 @@ def parse_args():
                              'syslog, or a file path')
     parser.add_argument('--grpcMaxMessageLength', type=int,
                         dest='grpc_max_msg_len')
+    parser.add_argument('--functions-uri', dest='functions_uri',
+                        help='URI with IP Address and Port used to'
```
Comment (YunchuWang) [inline azure_functions_worker/main.py:30]: nit: type=str for these args

Comment (YunchuWang) [reply azure_functions_worker/main.py:30]: any of these fields are required? default length for max_msg_len arg?

Comment (pdthummar) [reply azure_functions_worker/main.py:29]: Updated type=str for args. These args are required to handle from worker side. Thanks

File: tests/unittests/test_main.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import unittest
+
+from azure_functions_worker.main import create_args_parser
+
+
+class TestMain(unittest.TestCase):
+
```
Comment (gavin-aguiar) [inline tests/unittests/test_main.py:7]: Have to tried mocking the sys.argv? 
Something like this.
```
@patch.object(sys, 'argv', ['xxx', '--host', 'localhost',])
    def test_parse_args(self):
        args = parse_args()
        self.assertEqual(args.host, 'localhost')
```

In this way you wont have to change the code in main.

File: azure_functions_worker/main.py [hunk 3]
```diff
@@ -26,7 +25,19 @@ def parse_args():
...
+    parser.add_argument('--functions-request-id', dest='functions_request_id',
+                        type=str, help='Request ID used for gRPC communication '
+                                       'with the Host.')
+    parser.add_argument('--functions-worker-id',
+                        dest='functions_worker_id', type=str,
+                        help='Worker ID assigned to this language worker.')
+    parser.add_argument('--functions-grpc-max-message-length', type=int,
+                        dest='functions_grpc_max_msg_len',
+                        help='Max grpc message length for Functions')
+    return parser
```
Comment (kshyju) [inline azure_functions_worker/main.py:40]: Just confirming, if the new args are not present, the parsing will still work for old arguments(--host, --port etc.). correct?

Comment (gavin-aguiar) [reply azure_functions_worker/main.py:40]: It should. If the new args are not present, it should be defaulted to None and the old args will be used.
@Prashant, add a unit test for this too.

File: tests/unittests/test_main.py (hunk 2 above)
Comment (pdthummar) [reply tests/unittests/test_main.py:8]: updated all tests.

**Review by gavin-aguiar — APPROVED** [review]

**Review by vrdmr — APPROVED** [review]

**Pull request merged**